core/src/main/scala/org/apache/spark/util/io/ChunkedByteBuffer.scala

```diff
@@ -84,6 +92,64 @@ private[spark] class ChunkedByteBuffer(var chunks: Array[ByteBuffer]) {
     }
   }
 
+  /**
+   * write to stream with zero copy if possible
+   */
+  def writeToStream(out: OutputStream): Unit = {
+    var buffer: Array[Byte] = null
+    val bufferLen = 1024 * 1024
+    writeBufferToDest(this, out.write)
+  }
```

Comment:
Why do we need this method ?
If not required, removing it will simplify `writeExternal`

Reply:
good catch. This method is in my first implementation and I actually don't need it any more. Removed it and also remove the `writeBufferToDest`